### PR TITLE
Safer Message reading and writing.

### PR DESCRIPTION
**Pull request by pasnox**
The Message::payload() has been hidden and now only used
by the Message's >> and << operators.
Those operators ensure compile time check on meta types and run time
check on the payload stream status.

### CODE REVIEW

File: common/message.h [hunk 1]
```diff
@@ -63,10 +63,50 @@ class GAMMARAY_COMMON_EXPORT Message
...
+    /** Write value to the payload.
+     *  This operator proxy over payload() allow to do:
+     *   - Compile time check on the metatype id
+     *   - Run time check on the stream status
+     */
+    template <typename T>
+    GammaRay::Message &operator<<(const T &value)
+    {
+        const int id = qMetaTypeId<T>();
+        const char *name = QMetaType::typeName(id);
```
Comment (vkrause) [inline common/message.h]: Maybe move those lines out of the good case path?

File: common/message.h [hunk 2]
```diff
@@ -63,10 +63,50 @@ class GAMMARAY_COMMON_EXPORT Message
...
+        if (payload().status() != QDataStream::Ok)
```
Comment (vkrause) [inline common/message.h]: Candidate for Q_UNLIKELY, these methods are on a hot path I think.

**Comment by krf** [comment]
Rest looks good to me, :+1:

**Comment by vkrause** [comment]
Tested this,  it seems too strict actually. It fails on containers with older Qt versions (prior to those being auto-registered as meta types), yet that has absolutely no impact on QDataStream (de)serialization. What's the reason behind enforcing meta type registration?

File: common/message.h (hunk 1 above)
Comment (pasnox) [inline common/message.h]: Done

File: common/message.h (hunk 2 above)
Comment (pasnox) [inline common/message.h]: Done

**Comment by pasnox** [comment]
I will try to check builds errors with oldest Qt.
Enforcing the metatype registration make sure we can send queued events for the used types.
I lost some time finding that one of the root of a broken message comes from an unseen deserialization error hidden in code which can also not be ours (yeah, you QVariantMap). Some QVariant were not (de)serializable and silently fails, just the stream status was updated, but we were not doing any check one that.

**Pull request merged**